packages/flutter/test/widgets/navigator_test.dart

```diff
@@ -1813,6 +1813,57 @@ void main() {
     expect(tickCount, 4);
   });
 
+  testWidgets('Route annouce correctly for first route and last route', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/57133.
+    Route<void> previousOfFirst = NotAnnounced();
+    Route<void> nextOfFirst = NotAnnounced();
+    Route<void> popNextOfFirst = NotAnnounced();
+    Route<void> firstRoute;
+
+    Route<void> previousOfSecond = NotAnnounced();
+    Route<void> nextOfSecond = NotAnnounced();
+    Route<void> popNextOfSecond = NotAnnounced();
+    Route<void> secondRoute;
+
+    final GlobalKey<NavigatorState> navigator = GlobalKey<NavigatorState>();
+    await tester.pumpWidget(
+      MaterialApp(
+        navigatorKey: navigator,
+        initialRoute: '/second',
+        onGenerateRoute: (RouteSettings settings) {
+          if (settings.name == '/') {
+            firstRoute = RouteAnnouncementSpy(
+              onDidChangeNext: (Route<void> next) => nextOfFirst = next,
+              onDidChangePrevious: (Route<void> previous) => previousOfFirst = previous,
+              onDidPopNext: (Route<void> next) => popNextOfFirst = next,
+              settings: settings,
+            );
+            return firstRoute;
+          }
+          secondRoute = RouteAnnouncementSpy(
+            onDidChangeNext: (Route<void> next) => nextOfSecond = next,
+            onDidChangePrevious: (Route<void> previous) => previousOfSecond = previous,
+            onDidPopNext: (Route<void> next) => popNextOfSecond = next,
+            settings: settings,
+          );
+          return secondRoute;
+        },
+      ),
+    );
+    await tester.pumpAndSettle();
+
+    expect(previousOfFirst, isNull);
+    expect(nextOfFirst, secondRoute);
+    expect(popNextOfFirst is NotAnnounced, isTrue);
```

Comment:
Here and elsewhere: use the isA matcher: https://api.flutter.dev/flutter/package-matcher_matcher/isA.html